homeassistant/components/sonos/__init__.py

```diff
@@ -92,17 +107,19 @@ def _stop_discovery(event):
             data.hosts_heartbeat()
             data.hosts_heartbeat = None
 
-    def _discovery(now=None):
+    def _discovery(now: datetime.datetime | None = None) -> None:
```

Comment:
I think this is typically written as `Optional[datetime.datetime]` from other mypy examples I've seen

Reply:
The current notation in this PR is the new notation and is correct.